sdk/logical/event.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+option go_package = "github.com/hashicorp/vault/sdk/logical";
+
+package logical;
+
+// EventPluginInfo contains data related to the plugin that generated an event.
+message EventPluginInfo {
+    // The type of plugin this event originated from, i.e., "auth" or "secrets.
+    string mount_class = 1;
+    // Unique ID of the mount entry, e.g., "kv_957bb7d8"
+    string mount_accessor = 2;
+    // Mount path of the plugin this event originated from, e.g., "secret/"
+    string mount_path = 3;
+    // Plugin name that this event originated from, e.g., "kv"
+    string plugin = 4;
+    // Plugin version of the plugin this event originated from, e.g., "v0.13.3+builtin"
+    string plugin_version = 5;
+    // Mount version that this event originated from, i.e., if KVv2, then "2". Usually empty.
+    string version = 6;
+}
+
+// EventData contains event data in a CloudEvents container.
+message EventData {
+    // EID identifies the event, cannot be an empty and is required. The
+    // combination of Source + EID must be unique. Events with the same Source +
```

Comment:
Where is source defined?

Reply:
I'll clarify, but "source" is part of the CloudEvents spec, so will probably be the specific Vault cluster.

Reply:
In that case, is source + EID enough? Plugins will generate their own event IDs IIUC, so a poorly implemented plugin could cause issues by not generating event IDs with enough entropy.

Reply:
We default to using a UUID. I can add some more documentation to the ID field it should only be set manually if the ID has enough entropy (and possibly is idempotent).